trino/client.py

```diff
@@ -887,7 +895,11 @@ def decorated(*args, **kwargs):
                 try:
                     result = func(*args, **kwargs)
                     if any(guard(result) for guard in conditions):
-                        handle_retry.retry(func, args, kwargs, None, attempt)
+                        if result.status_code == 429 and "Retry-After" in result.headers:
+                            handle_retry_sleep = _RetryAfterSleep(result.headers.get("Retry-After"))
```

Comment:
Add a method to convert header value to seconds. The header can be a point-in-time too. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After#syntax in which case this code will throw.

Reply:
I've added a method `_parse_retry_after_header` for parsing `Retry-After` header.